lib/index.js

```diff
@@ -345,13 +345,23 @@ exports.deepEqual = function (obj, ref, seen) {
 
 // Remove duplicate items from array
 
-exports.unique = function (array, key) {
+exports.unique = function (array, keys) {
+
+    if (keys && !Array.isArray(keys)) {
+        keys = [keys];
+    }
 
     var index = {};
     var result = [];
 
     for (var i = 0, il = array.length; i < il; ++i) {
-        var id = (key ? array[i][key] : array[i]);
+        var id = array[i];
+        if (keys) {
+            id = '';
+            for (var x = 0; x < keys.length; x++) {
+                id += array[i][keys[x]];
```

Comment:
array[i] doesn't change between iterations, but maybe v8 already optimizes that.